src/style-spec/function/convert.js

```diff
@@ -236,8 +236,8 @@ function appendStopPair(curve, input, output, isStep) {
 function getFunctionType(parameters, propertySpec) {
     if (parameters.type) {
         return parameters.type;
-    } else if (propertySpec.function) {
-        return propertySpec.function === 'interpolated' ? 'exponential' : 'interval';
+    } else if (propertySpec.expression) {
```

Comment:
Should this be asserting `propertySpec.expression`? We shouldn't be trying to create a function for something that doesn't support expressions.

Reply:
👍 this was a bad workaround for a flow error — will fix.